### ISSUE TITLE
XInput causes UWP projects to crash on mobile with .NET Native

### ISSUE THREAD

**Issue by Nezz** [mono/MonoGame#4073]
XInput works fine on desktop Windows 10, but causes an issue when the game is deployed to mobile. If .NET Native is not used (i.e. debug builds), it throws an exception:
```
Exception thrown: 'System.DllNotFoundException' in SharpDX.XInput.dll
```

A catch for this exception was added by @tomspilman:
https://github.com/mono/MonoGame/blob/develop/MonoGame.Framework/Input/GamePad.XInput.cs#L167

However, if .NET Native is used, the application simply fails to launch

```
The program '[5112] example.exe' has exited with code -1073741515 (0xc0000135) 'A dependent dll was not found'.
Activation of the Windows Store app 'example' failed with error 'The app didn't start'.
```

The issue was fixed by removing references to XInput (using GamePad.Default).

**Comment by Nezz** [comment]
@tomspilman Do you think there is a way to fix this without ditching GamePad support?

**Comment by nkast** [comment]
XInput has been replaced by a new API that works across all platforms.

https://msdn.microsoft.com/en-us/library/mt282476.aspx
https://msdn.microsoft.com/en-us/library/windows/apps/windows.gaming.input

"If your game needs to support gampad input and you have existing XInput code, you can continue to make use of XInput. XInput has been replaced by Windows.Gaming.Input for UWP, and if you're writing new input code you should use Windows.Gaming.Input instead of XInput."

"The Windows.Gaming.Input API replaces XInput and provides the same functionality with the following advantages over Xinput: 
◦Lower resource usage
◦Lower API call latency for retrieving input
◦The ability to work with more than 4 gamepads at once
◦The ability to access addition Xbox One gamepad features, such as the trigger vibration motors
◦The ability to be notified when controllers connect/disconnect via event instead of polling
◦The ability to attribute input to a specific user (Windows.System.User)"

**Comment by tomspilman** [comment]
We need to implement this new API under a new GamePad implementation for Win10.

Just make a copy of:

https://github.com/mono/MonoGame/blob/develop/MonoGame.Framework/Input/GamePad.XInput.cs

... rename it something like `GamePad.UWP.cs` and add it to the protobuild config for Win10.  Then gut and fill it out with the new API.  So far it seems like this new API is pretty compatible with XInput feature wise.

**Comment by Nezz** [comment]
I have no controllers to test it with, so if anyone fells like doing it I can test whether it still crashes on mobile.

**Comment by cra0zy** [comment]
I got this.

**Comment by cra0zy** [comment]
we may have a problem.... Windows.Gaming.Input.Gamepad can't get the state of the Guide button(aka. BigButton in Xna... who came up with this name....)

**Comment by theZMan** [comment]
You've never been able to read the Guide button from XInput its reserved for system use. At least on the 360. 

BigButton is a totally different controller that came with the SceneIt game - see http://www.amazon.com/Scene-It-Controller-Pack-Xbox-360/dp/B004FWTXDO

There's some hacks if you call XInput with a secret struct or you can get it using raw input see http://forums.tigsource.com/index.php?topic=26792.0

**Comment by cra0zy** [comment]
oh, did not know that, I usually work on Linux and there guide button is treated like normal controller buttons....

anyway almost done, only vibrations left.

**Comment by theZMan** [comment]
Is it mapped to BigButton on Linux? If so then that's incorrect per the XNA API. There should probably be an additional GamePadButtons.Guide added for non microsoft platforms.

**Comment by BrianPeek** [comment]
The other option is to have SharpDX P/Invoke against XInputUAP.dll for Win10 UWP projects...this DLL exists across the board and maps XInput calls to their Gamepad API equivalents under the hood.  I've done this successfully by hex-editing the SharpDX.XInput.dll and changing the text from xinput1_4.dll to xinputuap.dll (luckily the same number of characters :) ).

That said, using the Gamepad API for Win10 is probably the *right* choice...  I started working on it a while ago but never got back to it.  @cra0zy, happy to answer questions if you have them.

**Comment by cra0zy** [comment]
If anyone wants to test out the gamepad, the PR is up at: #4077

**Comment by mrhelmut** [comment]
Good to me @cra0zy ! Thx.

**Comment by tomspilman** [comment]
This looks to be fixed with #4077 being merged.
